testserver/server.py

```diff
@@ -0,0 +1,106 @@
+#!/usr/bin/python
+
+import threading 
+import socket
+import select
+
+
+def consume_socket_content(sock, chunks=65536, timeout=0.5):
+    content = ""
+    more_to_read = select.select([sock], [], [], timeout)[0]
+
+    while more_to_read:
+        new_content = sock.recv(chunks).decode("utf-8")
+
+        if len(new_content) == 0:
+            more_to_read = False # empty recv means the socket disconnected
+
+        else:
+            content += new_content 
+            # stop reading if no new data is received for a while 
+            more_to_read = select.select([sock], [], [], timeout)[0] 
```

Comment:
While we're here, we may want to refactor this whole block. Rather than the `if/else`, let's try:

```python
while more_to_read:
    new_content = sock.recv(chunks)
    if not new_content:
         break
    content += new_content
    more_to_read = select.select([sock], [], [], timeout)[0]
```